openmdao/core/driver.py

```diff
@@ -708,66 +734,60 @@ def record_iteration(self):
         if not self._rec_mgr._recorders:
             return
 
-        metadata = create_local_meta(self._get_name())
+        # Get the data to record (collective calls that get across all ranks)
+        opts = self.recording_options
+        filt = self._filtered_vars_to_record
 
-        # Get the data to record
-        data = {}
-        if self.recording_options['record_desvars']:
-            # collective call that gets across all ranks
-            desvars = self.get_design_var_values()
+        if opts['record_desvars']:
+            des_vars = self.get_design_var_values()
         else:
-            desvars = {}
+            des_vars = {}
 
-        if self.recording_options['record_responses']:
-            # responses = self.get_response_values() # not really working yet
-            responses = {}
+        if opts['record_objectives']:
+            obj_vars = self.get_objective_values()
         else:
-            responses = {}
+            obj_vars = {}
 
-        if self.recording_options['record_objectives']:
-            objectives = self.get_objective_values()
+        if opts['record_constraints']:
+            con_vars = self.get_constraint_values()
         else:
-            objectives = {}
+            con_vars = {}
 
-        if self.recording_options['record_constraints']:
-            constraints = self.get_constraint_values()
+        if opts['record_responses']:
+            # res_vars = self.get_response_values()  # not really working yet
```

Comment:
I thought that, if anything was ever going to use this "response" stuff, it was the DOE driver. Maybe it is time to strip that out of the API, particularly the recording part, which is already complicated. Doesn't affect this PR though.